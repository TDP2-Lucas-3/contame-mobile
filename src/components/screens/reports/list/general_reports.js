import React from 'react';
import {getReports} from '../../../../config/routes';
import ReportsList from './reports_list';

const GeneralReports = (props) => (
  <ReportsList {...props} reportsAPI={getReports()} />
);

export default GeneralReports;
