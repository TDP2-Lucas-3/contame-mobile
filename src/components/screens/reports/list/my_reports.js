import React from 'react';
import {getMyReports} from '../../../../config/routes';
import ReportsList from './reports_list';

const MyReports = (props) => (
  <ReportsList {...props} reportsAPI={getMyReports()} />
);

export default MyReports;
