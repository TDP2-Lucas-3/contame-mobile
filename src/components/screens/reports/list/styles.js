import {StyleSheet} from 'react-native';

export const styles = StyleSheet.create({
  voteIcon: {
    padding: 5,
    paddingLeft: 10,
    paddingRight: 10,
    width: 55,
    flexDirection: 'row',
    justifyContent: 'space-between',
    borderColor: 'white',
    alignItems: 'center',
  },
});
