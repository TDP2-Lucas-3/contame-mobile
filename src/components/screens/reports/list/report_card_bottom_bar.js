import React from 'react';
import {colors, Icon, Text} from 'react-native-elements';
import {View} from 'react-native';
import {styles} from './styles';

export const ReportCardBottomBar = (props) => {
  return (
    <View style={styles.voteIcon}>
      <Icon
        color={props.voteByUser ? colors.primary : 'black'}
        name={'thumb-up-alt'}
        type="material"
        size={15}
      />
      <Text>{props.votes}</Text>
    </View>
  );
};
