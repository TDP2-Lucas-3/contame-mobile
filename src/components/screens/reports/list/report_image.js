import {View} from 'react-native';
import {Badge, Card, Icon} from 'react-native-elements';
import {styles} from '../../../../styles/common';
import {verboseReportState} from '../../../../utils/verbose_report_names';
import React from 'react';

export const ReportImage = (props) => {
  console.log(props.image);
  return (
    <View style={styles.borderTopRadius_1}>
      {props.image ? (
        <Card.Image
          testID="card_image"
          source={{uri: props.image}}
          style={styles.borderTopRadius_1}>
          <Badge
            badgeStyle={styles.report_status_badge}
            status={'warning'}
            value={verboseReportState(props.state)}
          />
        </Card.Image>
      ) : (
        <View>
          <Icon
            name="image-not-supported"
            testID="default_icon"
            type="material"
            containerStyle={[
              styles.borderTopRadius_1,
              styles.report_default_icon_container,
            ]}
          />
          <Badge
            badgeStyle={styles.report_status_badge_no_icon}
            status={'warning'}
            value={verboseReportState(props.state)}
          />
        </View>
      )}
    </View>
  );
};
