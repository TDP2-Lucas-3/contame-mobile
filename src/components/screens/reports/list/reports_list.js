import React from 'react';
import {FlatList, Text, View} from 'react-native';
import {Icon} from 'react-native-elements';
import {getReports} from '../../../../config/routes';
import ReportDetails from './report_details';
import useAxios from 'axios-hooks';
import {styles} from '../../../../styles/common';

const ReportsList = ({navigation}) => {
  const [{data, loading}, refetch] = useAxios(getReports());

  return (
    <View style={styles.container}>
      <FlatList
        data={data}
        renderItem={({item}) => (
          <ReportDetails
            {...item}
            onPress={() =>
              navigation.navigate('ReportDetails', {reportId: item.id})
            }
          />
        )}
        refreshing={loading}
        onRefresh={refetch}
      />
      <Icon
        name="plus"
        type="font-awesome-5"
        color="white"
        containerStyle={styles.action_button}
        onPress={() => navigation.navigate('NewReport')}
      />
    </View>
  );
};

export default ReportsList;
