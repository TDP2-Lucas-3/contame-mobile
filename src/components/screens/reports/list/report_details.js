import moment from 'moment';
import React from 'react';
import {View, TouchableOpacity} from 'react-native';
import {Card, Text, Icon} from 'react-native-elements';
import {styles} from '../../../../styles/common';
import {ReportCardBottomBar} from './report_card_bottom_bar';
import {ReportImage} from './report_image';

const ReportDetails = (props) => {
  return (
    <TouchableOpacity onPress={props.onPress}>
      <Card containerStyle={styles.report_card_container}>
        <ReportImage image={props.images[0]} state={props.state} />
        <View style={styles.report_card_body}>
          <View style={styles.m_1}>
            <Text style={styles.report_card_title}>{props.title}</Text>
            <Text style={[styles.color_grey, styles.raleway, styles.font_14]}>
              {moment(props.creationDate).fromNow()}
            </Text>
          </View>
          <View style={styles.m_1}>
            <Icon name="chevron-right" size={30} color="white" />
          </View>
        </View>
        <ReportCardBottomBar
          voteByUser={props.voteByUser}
          votes={props.votes}
        />
      </Card>
    </TouchableOpacity>
  );
};

export default ReportDetails;
