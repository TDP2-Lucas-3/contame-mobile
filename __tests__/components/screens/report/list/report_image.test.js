import React from 'react';
import {shallow} from 'enzyme';
import {findByTestId} from '../../../../support/helpers';
import {ReportImage} from '../../../../../src/components/screens/reports/list/report_image';

describe('ReportImage', () => {
  let render;
  let props;

  beforeEach(() => {
    props = {
      image: 'some image data',
      state: 'REPORTED',
    };
    render = (appProps) => shallow(<ReportImage {...appProps} />);
  });

  it('renders without crashing', () => {
    expect(() => render(props)).not.toThrowError();
  });

  it('renders all values correctly', () => {
    expect(render(props)).toMatchSnapshot();
  });

  describe('when there are multiple images', () => {
    beforeEach(() => {
      props.image = 'image';
    });

    it('renders image', () => {
      expect(findByTestId(render(props), 'card_image')).toHaveProp(
        'source',
        expect.objectContaining({uri: props.image}),
      );
    });
  });

  describe('when there is no image', () => {
    beforeEach(() => {
      props.image = undefined;
    });

    it("doesn't render any image", () => {
      expect(findByTestId(render(props), 'card_image')).toHaveLength(0);
    });

    it('renders default icon', () => {
      expect(findByTestId(render(props), 'default_icon')).toHaveLength(1);
    });
  });
});
