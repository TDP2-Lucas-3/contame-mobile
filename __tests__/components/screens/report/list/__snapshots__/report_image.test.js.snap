// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReportImage renders all values correctly 1`] = `
<View
  style={
    Object {
      "borderTopLeftRadius": 10,
      "borderTopRightRadius": 10,
    }
  }
>
  <Themed.CardImage
    source={
      Object {
        "uri": "some image data",
      }
    }
    style={
      Object {
        "borderTopLeftRadius": 10,
        "borderTopRightRadius": 10,
      }
    }
    testID="card_image"
  >
    <Themed.Badge
      badgeStyle={
        Object {
          "margin": 10,
          "padding": 5,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      status="warning"
    />
  </Themed.CardImage>
</View>
`;
